shell/platform/android/io/flutter/embedding/android/FlutterView.java

```diff
@@ -112,6 +116,8 @@ public void onFirstFrameRendered() {
    *   <li>{@link #renderMode} defaults to {@link RenderMode#surface}.</li>
    *   <li>{@link #transparencyMode} defaults to {@link TransparencyMode#opaque}.</li>
    * </ul>
+   * {@code FlutterView} requires an {@code Activity} instead of a generic {@code Context}
```

Comment:
Should the signature be changed to actually take an Activity throughout?

Reply:
I thought the approach we were talking about last week was that we're keeping the PlatformViewController to have an ApplicationContext constructor but just that we're adding new APIs to it, something like onAttachToActivity. And you can only do presentation.show after you onAttachToActivity. But then FlutterView isn't involved in any of this. It's just between the fragment, which does flutterPluginRegistry.getPlatformViewsController.onAttachActivity(activity), and the PlatformViewsController.

Reply:
@mklim I tried that but compilation failed because apparently Android `View` subclasses need to declare exact replicas of superclass constructors (probably due to XML inflation concerns). So I backed out that change and left a comment.

@xster the reason this was a problem at all is because we can't use an application context. If you ever attempt to `show()` a dialog/presentation with an application context, the app will blow up. That's why the previous commit that used the application context had to be reverted, which sparked the conversation with the Android team, etc. So now we're doing what we have to do to give `PlatformViewsController` an `Activity`.